src/persist/src/location.rs

```diff
@@ -33,7 +34,7 @@ use crate::error::Error;
 /// Read-only requests are assigned the SeqNo of a write, indicating that all
 /// mutating requests up to and including that one are reflected in the read
 /// state.
-#[derive(Clone, Copy, Debug, PartialOrd, Ord, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, Debug, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize)]
```

Comment:
hmmm why? this might want to roundtrip through some existing proto

Reply:
I don't understand what roundtripping through an existing proto means here. Can I get some more context? The "why" is just that I want to make `SeqNo` exchange-able, though sounds like there's a better strategy.

Reply:
@danhhz if we have a newtype around `SeqNo` (as discussed [here](https://github.com/MaterializeInc/materialize/pull/14177/files#r942817740)). Do you think that newtype should just have a yolo serde implementation or use protos? It seems we never want to store that in state so wouldn't need a proto.

Reply:
I was assuming that Serialize and Deserialize were being added here because you wanted SeqNo to impl ExchangeData. my "hmmm why?" was asking you to point out where that was used (I could have dug this up myself, but I was very quickly scanning the PR for the initial feedback). "this might want to roundtrip through some existing proto" was a poorly explained version of "wanting SeqNo to directly be exchange-able is a code smell to me" :). IMO serde (especially combined with bincode, which is what timely does with it) gives pretty bad tooling for actually reasoning about forward and backward compatibility. this is why persist does the dance where we use the thinnest possible FooSerde shim to proxy the actual encoding/decoding to protobuf

I realize now that I'm thinking more about it that not only does SeqNo need to be sent from operator stage 1 -> 2 as part of the batch, it also needs to be sent from stage 2 -> 3. which means I think we need a new copy of the FooSerde+FooProto pattern for the SeqNo/LeasedSeqNo data being sent over this dataflow edge

Reply:
tysm for the explanation! will include this change, natch

Reply:
Done in commit w/ message `persist: add encapsulation around SeqNo leases`